### ISSUE TITLE
Any plan to support new version of VRM?

### ISSUE THREAD

**Issue by is0263rf** [tattn/VRMKit#27]
このリポジトリはVRM 0.xに対応していて[VRM 1.0](https://vrm.dev/vrm1/)には対応していないようです。
VRM 1.0に対応する予定はありますか?

**Comment by tattn** [comment]
Issueありがとうございます！

VRM 1.0に対応するモチベーションはありますが、現在対応するための時間がなく、対応できていません。
VRM 1.0対応のpull requestは受け付けています。

[en]
Thank you for this issue!

I'm motivated to support VRM 1.0, but currently don't have the time to provide the support.
I welcome pull requests for VRM 1.0 support.

**Comment by is0263rf** [comment]
ありがとうございます。実装と仕様書を理解してみて、行けそうならPull Requestを送ります。

**Comment by c4b4d4** [comment]
VRM 1.0 レンダリングは現在主流ですか? そうでない場合、その必要性は何ですか? レンダリングはできず、「分析」することしかできないようですが?

Is Rendering VRM 1.0 a thing now? If not, what is needed? I see that only "parsing" is possible, but not rendering them?

**Comment by tattn** [comment]
@c4b4d4 
Yes, VRM 1.0 is becoming mainstream, but VRM 0.x models are still widely used.
To render VRM 1.0, it's necessary to adapt the parsed results to the rendering system of Apple platforms (SceneKit), which will take lots of time to support.
Currently, VRMKit is in the middle of supporting VRM 1.0.

**Comment by tattn** [comment]
Thank you for your interest!

I'm currently busy with my main job, so I haven't been able to start working on this myself. I'm sorry.
The most challenging part is mapping the parsed data to the various classes in SceneKit.

The most important task is to make it possible to create a `VRMNode` by passing a `VRM1` object, as it is currently feasible to load a `VRM` as a `VRMNode`. Modifying this part requires changing many affected codes.

https://github.com/tattn/VRMKit/blob/f45e1ea/Sources/VRMSceneKit/VRMSceneLoader.swift#L36

*VRM refers to VRM 0.x model data, and VRM1 refers to VRM 1.x model data.*

**Comment by is0263rf** [comment]
Hi, @c4b4d4 . Actually, I have successfully rendered a VRM 1.0 file. But I haven't sent a pull request because my implementation has some shortcomings.

https://github.com/is0263rf/VRMKit/tree/development

For example, I haven't implemented Expression (BlendShape).

https://vrm.dev/vrm1/expression/

I don't know if my implementation will be useful to you. I look forward to your pull request to the tattn's repository.

**Comment by tattn** [comment]
This issue has been resolved by https://github.com/tattn/VRMKit/releases/tag/0.8.0, so I’ll close it.
Please open a new issue if there are any remaining problems.
